src/charm_state.py

```diff
@@ -55,12 +108,27 @@ def from_charm(cls, charm: CharmBase) -> "CharmState":
 
         Return:
             The CharmState instance created by the provided charm.
+
+        Raises:
+            CharmConfigInvalidError: if the charm configuration is invalid.
         """
         keepalive = charm.config.get("webserver_keepalive")
         timeout = charm.config.get("webserver_timeout")
         workers = charm.config.get("webserver_workers")
         threads = charm.config.get("webserver_threads")
+        flask_config = {
+            k.removeprefix("flask_"): v for k, v in charm.config.items() if k.startswith("flask_")
+        }
+        try:
+            valid_flask_config = FlaskConfig(**flask_config)  # type: ignore
```

Comment:
What is the typing problem here please?

Reply:
pydantic model's constructor is typed with each individual field as keyword arguments, which is not compatible with dict unpacking.

Reply:
I added a comment for that, thanks!